internal/corazawaf/rule.go

```diff
@@ -282,16 +288,18 @@ func (r *Rule) doEvaluate(phase types.RulePhase, tx *Transaction, cache map[tran
 			matchedValues = append(matchedValues, matchedChainValues...)
 			nr = nr.Chain
 		}
-		// we need to add disruptive actions in the end, otherwise they would be triggered without their chains.
-		if tx.RuleEngine != types.RuleEngineDetectionOnly {
-			tx.WAF.Logger.Debug("[%s] [%d] Disrupting transaction by rule %d", tx.id, rid, r.ID_)
-			for _, a := range r.actions {
-				if a.Function.Type() == rules.ActionTypeDisruptive || a.Function.Type() == rules.ActionTypeFlow {
-					tx.WAF.Logger.Debug("[%s] [%d] Evaluating action %s for rule %d", tx.id, rid, a.Name, r.ID_)
-					a.Function.Evaluate(r, tx)
-				}
-			}
 
+		for _, a := range r.actions {
+			if a.Function.Type() == rules.ActionTypeFlow {
+				// Flow actions are evaluated also if the rule engine is set to DetectionOnly
+				tx.WAF.Logger.Debug("[%s] [%d] Evaluating flow action %s for rule %d", tx.id, rid, a.Name, r.ID_)
+				a.Function.Evaluate(r, tx)
+			} else if a.Function.Type() == rules.ActionTypeDisruptive && tx.RuleEngine == types.RuleEngineOn {
+				// We need to add disruptive actions in the end, otherwise they would be triggered without their chains.
```

Comment:
Can we remove this comment? See below

Reply:
Looking[ at the initial commit](https://github.com/corazawaf/coraza/blame/d798b28fbf16dffdb31ba908ea95561ee30021ab/pkg/waf/rule.go#L120), I think it may just be related to the fact that they are evaluated after the whole chain (not like non disruptive actions). So we should be okay just removing it. [This comment](https://github.com/corazawaf/coraza/pull/684/files#diff-32bf0985de09d747ed6a6d8400fa4e02e61e15b9cb67b3da4623c050bf1306ccR279) is already rephrasing the same concept.


cc @jptosso